src/gui/guiScrollBar.cpp

```diff
@@ -274,10 +274,10 @@ void GUIScrollBar::setPosRaw(const s32 &pos)
 	s32 thumb_min = 0;
 
 	if (is_horizontal) {
-		thumb_min = RelativeRect.getHeight();
+		thumb_min = std::min(RelativeRect.getHeight(), RelativeRect.getWidth() / 2);
```

Comment:
What's the reasoning behind this (and the change below)? It looks like it should enforce a certain minimum "ratio" of the thumb; where can I see this in action?

Reply:
usually, the scrollbar width is used as the thumb's minimum height

however, if your scrollbar is wider than tall, that obviously results in a too tall thumb, making the scrollbar unusable

therefore, this change makes sure that the thumb's minimum height (not actual height) doesn't exceed the scrollbar's height divided by two

(assuming a vertical scrollbar)

----

you can test this with `/test_formspec` from Devtest, "Styles" tab. choose a small window size, look at the textareas.